src/PSR7/Validators/CookiesValidator/RequestCookieValidator.php

```diff
@@ -59,22 +57,40 @@ private function checkRequiredCookies(Cookies $cookies, OperationAddress $addr)
     /**
      * @throws InvalidCookies
      */
-    private function checkCookiesAgainstSchema(RequestInterface $request, OperationAddress $addr, Cookies $cookies) : void
+    private function checkCookiesAgainstSchema(RequestInterface $request, OperationAddress $addr, array $cookies) : void
     {
         $validator = new SchemaValidator($this->detectValidationStrategy($request));
 
-        foreach ($cookies->getAll() as $cookie) {
-            /** @var Cookie $cookie */
-            if (! isset($this->specs[$cookie->getName()])) {
+        foreach ($cookies as $cookieName => $cookie) {
+            if (! isset($this->specs[$cookieName])) {
                 continue;
             }
 
-            $parameter = SerializedParameter::fromSpec($this->specs[$cookie->getName()]);
+            $parameter = SerializedParameter::fromSpec($this->specs[$cookieName]);
             try {
-                $validator->validate($parameter->deserialize($cookie->getValue()), $parameter->getSchema());
+                $validator->validate($parameter->deserialize($cookie), $parameter->getSchema());
             } catch (SchemaMismatch $e) {
-                throw InvalidCookies::becauseValueDoesNotMatchSchema($cookie->getName(), $cookie->getValue(), $addr, $e);
+                throw InvalidCookies::becauseValueDoesNotMatchSchema($cookieName, $cookie, $addr, $e);
             }
         }
     }
+
+    /**
+     * @return string[]
+     */
+    private function getCookiesFromMessage(MessageInterface $message): array
+    {
+        // Needed in case it is an array of cookies.
+        $cookieString = implode('; ', $message->getHeader('Cookie'));
```

Comment:
Can't you use `$message->getHeaderLine()`?

Reply:
That uses `implode` with `, ` We don't want that here. So it s the same as that one only glue is different.